staging/src/k8s.io/kube-aggregator/pkg/apiserver/openapi_controller.go

```diff
@@ -0,0 +1,90 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package apiserver
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/golang/glog"
+
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/util/workqueue"
+)
+
+type OpenAPIAggregationManager interface {
+	UpdateApiServiceSpec(apiServiceName string) (changed, deleted bool, err error)
+}
+
+type OpenAPIAggregationController struct {
+	openAPIAggregationManager OpenAPIAggregationManager
+	queue                     workqueue.RateLimitingInterface
+}
+
```

Comment:
By this struct you mean the controller? no it does not. `openAPIAggregator` does and I added a type assertion there.